subscriber/synapse-subscriber-kafka/src/main/java/io/americanexpress/synapse/subscriber/kafka/annotation/KafkaErrorHandler.java

```diff
@@ -0,0 +1,32 @@
+package io.americanexpress.synapse.subscriber.kafka.annotation;
+
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.stereotype.Component;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * {@code KafkaErrorHandler} annotation is used as annotate a class as a kafka subscriber error handler.
+ * This annotation serves as a specialization of @Component, allowing for implementation classes
+ * to be auto-detected through classpath scanning.
+ *
+ * @author Krishna Kuchikulla
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Component
+public @interface KafkaErrorHandler {
```

Comment:
🤔 why was an custom annotation added

Reply:
this annotation is to indicate that a class is used as errorHandler and is registered in kafka subscriber config